libraries/RC_Channel/RC_Channel.cpp

```diff
@@ -1250,6 +1251,34 @@ bool RC_Channel::do_aux_function(const aux_func_t ch_option, const AuxSwitchPos
         break;
     }
 
+    case AUX_FUNC::ARM_EMERGENCY_STOP: {
+        switch (ch_flag) {
+           case AuxSwitchPos::HIGH:
+              SRV_Channels::set_emergency_stop(false);
+              AP::arming().arm(AP_Arming::Method::AUXSWITCH, true);
```

Comment:
I wounder what the behavior should be if arming fails. Should E-Stop be left off or re-engaged?

Reply:
if left re-engaged, then you cant arm from another method...thinks its best left disengaged if switch is high...its really the union of a arm switch and an inverted estop switch...